test/react/javascript.jsx

```diff
@@ -0,0 +1 @@
+module.exports = 'react';
```

Comment:
I'm not super familiar with jsx, but is this a valid jsx file? I have never seen one with module.exports in it before.

Reply:
This is invalid JSX because we do not have Webpack to compile it, but I just need to support importing the .jsx file "out of the box".

Reply:
Gotcha. So these tests are for the "out of the box" experience, so just want to have something a normal user would have and experience so we can make sure it keeps working.

Reply:
Yes, React does not work with it, but this test only checks the file extension, not the code in the file. Imagine my app importing JSX files, but this is not a React-app.

Reply:
Sure, but inevitably folks are going to try to use this and contact me for support. I just need to make sure that the tests test what the real folks are going to try and do with it. I get issues already multiple times with folks who do not even know what "fs" is in Node.js, so I just need enough information and tests to be prepared for the issues that will pop up is all.

Reply:
I checked this in create-react-app and it does not work (because browser is not supports node's fs), but I want to not specify `filter: /^([^.].*)\.(j|t)sx?(on)?$/` each time, and keep it out of the box

Reply:
So if it doesn't work, then should it be added here? I'm a little confused on your response.

Reply:
I need support for these file extensions, not React support, so I added it here

Reply:
Gotcha. So can you explain to me how you will use it? I'm sure once this lands folks will start to try and use it, but there is no information on how to use it in this PR, so I don't have enough information to be able to support this feature yet.

Reply:
Okay.. I just need to support .jsx, .tsx and .ts file extensions, not React or TypeScript

Reply:
That's fine. I'm not familiar with any of this, so this is just what I need help on. So maybe this will help:

I am a user here who has a jsx file, and I use `require-all` to point to a directory that has it inside, but then what happens? If the file in this test is not a valid `jsx` file what would one look like? And that would just then work "out of the box" with `require-all`? If so, why not just put a valid `jsx` file in the tests here?

Reply:
I can't to put valid `jsx`, because we need the bundler like Webpack or Babel. It's not important for us to know what is in this file, the main thing is that the importer sees its extension. It PR is not resolves problems with React, it just adding .jsx file extension support.

Reply:
Sure, but folks may have problems and because this module is suddenly going to be trying to load these files, I need to have enough information to be able to support this. Part of putting a module out there also means people are going to open PRs and issues about the different features in the module. So I have no idea even myself how to use what you've added here, and so I don't know how I could possibly help anyone else. I'm just asking for how to make use of this new feature so I can understand and then help those who open issues about it in the future.

Reply:
I understood. Let's take a situation - we have such a directory:
`` ``
> foo
- bar.jsx
- styles.js
`` ``
and `require-all` will import the bar and styles together if it has not been imported before, and users may have problems. Then I will do PR not here, but in the repository, which depends on this package.